renderdoc/driver/d3d12/d3d12_common.h

```diff
@@ -75,15 +75,21 @@ inline std::string DecodeMarkerString(UINT Metadata, const void *pData, UINT Siz
 {
   std::string MarkerText = "";
 
+  // We have may appended a space to the marker string - see D3D12MarkerRegion::Begin
+  // If we encounter this extra space, remove it.
   if(Metadata == PIX_EVENT_UNICODE_VERSION)
   {
     const wchar_t *w = (const wchar_t *)pData;
-    MarkerText = StringFormat::Wide2UTF8(std::wstring(w, w + Size));
+    MarkerText = StringFormat::Wide2UTF8(std::wstring(w, w + Size / sizeof(wchar_t)));
+    if(!MarkerText.empty() && MarkerText.back() == ' ')
```

Comment:
I think we want to check against `(MarkerText.back() == ' ' || MarkerText.back() == 0x0)` here, in case someone passes what PIX expects - the string size including the NULL terminator.